src/lambda/vue/samInvoke.ts

```diff
@@ -0,0 +1,501 @@
+    )
+}
+
+export interface SamInvokeVueState {
+    launchConfig: MorePermissiveAwsSamDebuggerConfiguration
+    payload: { value: string; errorMsg: string }
+}
+
+export interface MorePermissiveAwsSamDebuggerConfiguration extends AwsSamDebuggerConfiguration {
+    invokeTarget: {
+        target: 'template' | 'api' | 'code'
+        templatePath: string
+        logicalId: string
+        lambdaHandler: string
+        projectRoot: string
+    }
+}
+
+export interface LoadSamLaunchConfigResponse {
+    command: 'loadSamLaunchConfig'
+    data: {
+        launchConfig: AwsSamDebuggerConfiguration
+    }
+}
+
+export interface GetSamplePayloadResponse {
+    command: 'getSamplePayload'
+    data: {
+        payload: string
+    }
+}
+
+export interface GetTemplateResponse {
+    command: 'getTemplate'
+    data: {
+        template: string
+        logicalId: string
+    }
+}
+
+export interface SamInvokerBasicRequest {
+    command: 'loadSamLaunchConfig' | 'getSamplePayload' | 'getTemplate'
+}
+
+export interface SamInvokerLaunchRequest {
+    command: 'saveLaunchConfig' | 'invokeLaunchConfig'
+    data: {
+        launchConfig: AwsSamDebuggerConfiguration
+    }
+}
+
+export type SamInvokerRequest = SamInvokerBasicRequest | SamInvokerLaunchRequest
+export type SamInvokerResponse = LoadSamLaunchConfigResponse | GetSamplePayloadResponse | GetTemplateResponse
+
+async function handleFrontendToBackendMessage(
+    message: SamInvokerRequest,
+    postMessageFn: (response: SamInvokerResponse) => Thenable<boolean>,
+    destroyWebviewFn: () => any,
+    context: ExtContext
+): Promise<any> {
+    switch (message.command) {
+        case 'loadSamLaunchConfig':
+            loadSamLaunchConfig(postMessageFn)
+            break
+        case 'getSamplePayload':
+            getSamplePayload(postMessageFn)
+            break
+        case 'getTemplate':
+            getTemplate(postMessageFn)
+            break
+        case 'saveLaunchConfig':
+            saveLaunchConfig(message.data.launchConfig)
+            break
+        case 'invokeLaunchConfig':
+            invokeLaunchConfig(message.data.launchConfig, context)
+            break
+    }
+}
+
+/**
+ * Open a quick pick containing the names of launch configs in the `launch.json` array.
+ * Filter out non-supported launch configs.
+ * Call back into the webview with the selected launch config.
+ * @param postMessageFn
+ */
+async function loadSamLaunchConfig(postMessageFn: (response: SamInvokerResponse) => Thenable<boolean>): Promise<void> {
+    // TODO: Find a better way to infer this. Might need another arg from the frontend (depends on the context in which the launch config is made?)
+    const workspaceFolder = vscode.workspace.workspaceFolders?.length ? vscode.workspace.workspaceFolders[0] : undefined
+    if (!workspaceFolder) {
+        // TODO Localize
+        vscode.window.showErrorMessage(localize('AWS.lambda.form.noFolder', 'No workspace folder found.'))
+        return
+    }
```

Comment:
similar pattern is found in `makeConfig()` https://github.com/aws/aws-toolkit-vscode/blob/3645a41372e5800f70b4d7dbcbf2770f7ac0868b/src/shared/sam/debugger/awsSamDebugger.ts#L318-L326

this should be generalized and moved to a common function, `src/shared/utilities/workspaceUtils.ts:getWorkspaceFolder()`